lib/KCard/__test__/KCard.spec.js

```diff
@@ -0,0 +1,76 @@
+import { createLocalVue, mount } from '@vue/test-utils';
+import VueRouter from 'vue-router';
+import KCard from './../index.vue';
+
+const localVue = createLocalVue();
+localVue.use(VueRouter);
+const router = new VueRouter();
+
+describe('index.vue', () => {
+  const cardFocusMock = jest.fn();
+
+  // beforeEach(() => {
+  const wrapper = mount(KCard, {
+    localVue,
+    router,
+    propsData: {
+      title: 'Sample Title',
+      headingLevel: 4,
+      to: { path: '/some-link' },
+      layout: 'horizontal',
+      thumbnailDisplay: 'small',
+      thumbnailSrc: 'https://example.com/image.jpg',
+    },
+    slots: {
+      aboveTitle: '<p>This is the default content</p>',
+      title: '<li>This is the default title</li>',
+      belowTitle: '<i>This is the default description</i>',
+      footer: '<b>This is the default footer</b>',
+    },
+    mocks: {
+      cardFocus: cardFocusMock,
+    },
+  });
+  // });
+
+  it('renders passed header level', () => {
+    expect(wrapper.find('h4').exists()).toBe(true);
+  });
+
+  //testing focus and hover events
+  it('renders data on focus', () => {
+    let emittedData;
+    wrapper.vm.$on('focus', data => {
+      emittedData = data;
+    });
+
+    expect(emittedData).toBeUndefined();
+  });
+
+  it('emits data on hover', () => {
+    let hoverData;
+    wrapper.vm.$on('hover', data => {
+      hoverData = data;
+    });
+
+    expect(hoverData).toBeUndefined();
+  });
+
+  //testing slots
+  it('renders slotted content', () => {
+    expect(wrapper.find('[slot="aboveTitle"]').exists()).toBe(false);
```

Comment:
Did you mean to test for the presence of the slot? If so, you'd need to say `toBe(true)`. The test, as implemented now, confirms that the slot _doesn't exist_ rather than that it does.

I think that lines 65-68 are sufficient to confirm that slot content gets displayed.

Reply:
I was trying to make sure that the card does not emit data by default and only emits data when it receives focus.